src/vs/editor/browser/core/editorState.ts

```diff
@@ -162,4 +163,18 @@ export class StableEditorScrollState {
 			editor.setScrollTop(visiblePositionScrollTop + this._visiblePositionScrollDelta);
 		}
 	}
+
+	public restoreRelativeVerticalPositionOfCursor(editor: ICodeEditor, shouldPushUndoStop: boolean = false): void {
+		const currentCursorPosition = editor.getPosition();
+
+		if (!this._cursorPosition || !currentCursorPosition) {
+			return;
+		}
+
+		const offset = editor.getTopForLineNumber(currentCursorPosition.lineNumber) - editor.getTopForLineNumber(this._cursorPosition.lineNumber);
+		if (shouldPushUndoStop) {
```

Comment:
I don't think `pushUndoStop` belongs here. This class should not do that.

The undo stop does not record scroll positions, so any caller of `restoreRelativeVerticalPositionOfCursor` can call `editor.pushUndoStop` after `editor.setScrollTop` and there will be the same effect.

Reply:
@alexdima 
Thank you. I fixed.